### PR TITLE
feat: introduce the hb nodes package to monorepo

### PR DESCRIPTION
**Pull request by ShragaUser**
# HELPER CREATED BY GPT 

# Add HoneyBook Nodes Support

This PR introduces support for HoneyBook nodes, expanding our workflow capabilities with new functionalities. 

### Changes
- **Prettier Ignore Updates**: Added exclusion for HoneyBook test files in the `.prettierignore`.
- **Package.json Adjustments**:
  - Added `n8n-nodes-honeybook` as a workspace dependency across various packages, ensuring seamless integration and development within our workspace environment.
- **Core Package Updates**:
  - Introduced `n8n-nodes-honeybook` as a peer dependency in the core package, highlighting its importance and interoperability within the core functionalities of n8n.
- **HoneyBook Node Implementation**:
  - Developed a comprehensive setup for HoneyBook nodes, including ESLint configurations, README documentation, credentials implementation, gulp tasks for asset management, and jest configurations for testing.
  - Implemented the `MovePipelineStage` node as part of the HoneyBook nodes, providing users with the ability to interact with HoneyBook's API for moving pipeline stages, thereby enhancing the automation capabilities within HoneyBook's ecosystem.
- **Utilities and Helpers**:
  - Added utility functions to facilitate communication with HoneyBook's API, encapsulating request handling and error management to streamline node interactions with the API.

### CODE REVIEW

**Comment by ShragaUser** [comment]
to run with new setup you need to run
```bash
pnpm install
export N8N_CUSTOM_EXTENSIONS=<N8N_REPO_LOCATION_ON_LOCAL_COMPUTER>/packages/n8n-nodes-honeybook/dist
export N8N_COMMUNITY_PACKAGES_ENABLED=true
pnpm run dev # or pnpm run start if you already ran build
```

at build time we will build the package directly to ~/.n8n/nodes directory where it is loaded automatically as a community node, we do not have to "enable community nodes" to enable this.

File: packages/n8n-nodes-honeybook/package.json [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    "format": "prettier --write . --ignore-path ../../.prettierignore",
+    "lint": "eslint . --quiet && node ./scripts/validate-load-options-methods.js",
+    "lintfix": "eslint . --fix",
+    "watch": "tsc-watch -p tsconfig.build.json --onCompilationComplete \"tsc-alias -p tsconfig.build.json\" --onSuccess \"pnpm n8n-generate-ui-types\"",
+    "test": "jest"
+  },
+  "files": [
+    "dist"
+  ],
+  "n8n": {
```
Comment (ShragaUser) [inline packages/n8n-nodes-honeybook/package.json:22]: add all node and credential info here for them to be exported

File: packages/n8n-nodes-honeybook/README.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
+# Package for HB nodes
```
Comment (ShragaUser) [inline packages/n8n-nodes-honeybook/README.md:1]: This README adds information on how to add packages and install in local settings 🙏

File: packages/n8n-nodes-honeybook/package.json [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+    "credentials": [
+      "dist/credentials/HoneyBookApi.credentials.js"
+    ],
+    "nodes": [
+      "dist/nodes/HoneyBookTrigger/HoneyBookTrigger.node.js",
+      "dist/nodes/HoneyBookTrigger/HoneyBookAction.node.js"
```
Comment (ShragaUser) [inline packages/n8n-nodes-honeybook/package.json:28]: This path needs to be updated I think it's wrong @thewizarodofoz

Comment (thewizarodofoz) [reply packages/n8n-nodes-honeybook/package.json:28]: oh funny - apparently n8n does not use this. it loads all the nodes in the nodes folder

**Pull request merged**